src/autoscaler/metricscollector/cmd/metricscollector/metricscollector_test.go

```diff
@@ -21,22 +32,74 @@ var _ = Describe("MetricsCollector", func() {
 		runner = NewMetricsCollectorRunner()
 	})
 
-	JustBeforeEach(func() {
-		runner.Start()
-	})
-
 	AfterEach(func() {
 		runner.KillWithFire()
 	})
 
-	It("should start", func() {
-		Consistently(runner.Session).ShouldNot(Exit())
+	Context("when the metricscollector acquires the lock", func() {
+		BeforeEach(func() {
+			runner.StartWithoutCheck()
+			Eventually(runner.Session.Buffer, 2*time.Second).Should(gbytes.Say(runner.acquiredLockCheck))
+		})
+
+		It("should start", func() {
+			Eventually(runner.Session.Buffer(), 2*time.Second).Should(gbytes.Say(runner.startCheck))
+			Consistently(runner.Session).ShouldNot(Exit())
+		})
+	})
+
+	Context("when the metricscollector loses the lock", func() {
+		BeforeEach(func() {
+			runner.StartWithoutCheck()
+			Eventually(runner.Session.Buffer, 2*time.Second).Should(gbytes.Say(runner.acquiredLockCheck))
+			Eventually(runner.Session.Buffer(), 2*time.Second).Should(gbytes.Say(runner.startCheck))
```

Comment:
Function Eventually should be passed function Buffer not Buffer(). Please check it is at other spots as well.